gitlab-pipeline/stage/init.yml

```diff
@@ -59,6 +59,19 @@ init:workspace:
           git checkout -f -b pr ${clone_rev}
         ) || return 1
 
+        (
+          cd $clone_path
+          # This is a no-op for repositories that don't have submodules. The `--depth=1` needs
+          # server support, and not all sources we clone submodules from support that, so let's try
+          # first, to get as much space saving as we can, but fall back on full clone.
+          git submodule update --init --depth=1 ||
```

Comment:
Hmm, have we not had issues with the depth before, if the sha we are looking for is at a deeper depth? :thinking:  I might be wrong here though.

Reply:
The problem is specific to submodules I believe. If you are lucky, it can work, if the submodule is at a SHA which is exactly where a branch or tag also is. But often, it isn't, and then it first clones using `--depth=1`, but then it still needs to fetch the specific commit you need, and not all servers support fetching direct SHAs, without a branch reference. Github didn't support this either in the past; they added support a couple of years ago.

The specific place which is a problem for us is when cloning gitdm inside the integration repository.